gtsam/slam/EssentialMatrixFactor.h

```diff
@@ -420,4 +417,104 @@ class EssentialMatrixFactor4
 };
 // EssentialMatrixFactor4
 
+/**
+ * Binary factor that optimizes for E and two calibrations Ka and Kb using the
+ * algebraic epipolar error (Ka^-1 pA)'E (Kb^-1 pB). The calibrations are
+ * assumed different for the two images. Don'tt use this factor with same
+ * calibration unknown, as Jacobians will be incorrect...
```

Comment:
can we check this? do you mean the case where keyKa == keyKb? we can maybe assert they're different in constructor?

Reply:
but if you mean "do not use this with different keys for calibrations that should be the same", yeah, that is a good comment to have.

Reply:
I added a test asserting that 
```c++
    factor4.evaluateError(trueE, trueK, nullptr, &H0);
    factor5.evaluateError(trueE, trueK, trueK, nullptr, &H1, &H2);
    EXPECT(assert_equal(H0, H1 + H2, 1e-9));
```